From 91076a4c05182a80043b59a90f977ada30e23e6b Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Wed, 26 Nov 2008 10:16:35 +0800
Subject: [PATCH] ENGR00100339: add wakeup source from kbd and tsc.

Add Keyboard(MPR084) and Touchscreen(tsc2007) as
wake up irq source.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/input/keyboard/mpr084.c     |    1 +
 drivers/input/touchscreen/tsc2007.c |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/input/keyboard/mpr084.c b/drivers/input/keyboard/mpr084.c
index 4b56b0d..8a82c24 100644
--- a/drivers/input/keyboard/mpr084.c
+++ b/drivers/input/keyboard/mpr084.c
@@ -272,6 +272,7 @@ static int mpr084_driver_register(struct mpr084_data *data)
 				IRQF_TRIGGER_FALLING, DRIVER_NAME, data);
 		if (!ret) {
 			init_completion(&data->kpirq_completion);
+			set_irq_wake(data->kpirq, 1);
 		} else {
 			printk(KERN_ERR "%s: cannot grab irq %d\n",
 			       __func__, data->kpirq);
diff --git a/drivers/input/touchscreen/tsc2007.c b/drivers/input/touchscreen/tsc2007.c
index 0896b7e..017f261 100644
--- a/drivers/input/touchscreen/tsc2007.c
+++ b/drivers/input/touchscreen/tsc2007.c
@@ -298,6 +298,7 @@ static int tsc2007_driver_register(struct tsc2007_data *data)
 		if (!ret) {
 			printk(KERN_INFO "%s: Registering Touchscreen device\n",
 			       __func__);
+			set_irq_wake(data->penirq, 1);
 		} else {
 			printk(KERN_ERR "%s: Cannot grab irq %d\n",
 			       __func__, data->penirq);
-- 
1.5.4.4

